extensions/ap4k/deployment/src/main/java/io/quarkus/ap4k/deployment/Ap4kProcessor.java

```diff
@@ -0,0 +1,171 @@
+package io.quarkus.ap4k.deployment;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.ServiceLoader;
+
+import javax.inject.Inject;
+
+import org.jboss.jandex.AnnotationInstance;
+import org.jboss.jandex.AnnotationValue;
+import org.jboss.jandex.DotName;
+import org.jboss.jandex.Index;
+
+import io.ap4k.Generator;
+import io.ap4k.Session;
+import io.ap4k.SessionWriter;
+import io.ap4k.processor.SimpleFileWriter;
+import io.ap4k.project.Project;
+import io.quarkus.deployment.BuildInfo;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.ApplicationIndexBuildItem;
+import io.quarkus.deployment.builditem.BuildInfoBuildItem;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.deployment.builditem.GeneratedResourceBuildItem;
+
+class Ap4kProcessor {
+
+    @Inject
+    BuildProducer<GeneratedResourceBuildItem> generatedResourceProducer;
+
+    @Inject
+    BuildProducer<FeatureBuildItem> featureProducer;
+
+    @BuildStep
+    public void build(ApplicationIndexBuildItem applicationIndexBuildItem, BuildInfoBuildItem buildInfoBuildItem) {
+
+        // the resources that ap4k's execution will result in, will later on be written
+        // by quarkus in the wiring-classes directory
+        // the location is needed in order to properly support s2i build triggering
+        final Path rootPath = Paths.get(buildInfoBuildItem.getBuildInfo().getWiringClassesDir());
+        if (rootPath.toString().isEmpty()) { //this is done to ap4k from failing when the necessary input has not been provided (during tests for example)
```

Comment:
s/this is done to ap4k from failing/this is done to **prevent** ap4k from failing/ ?

Reply:
Damn... I really really suck at writing!